packages/data-point/lib/helpers/type-check-function-reducers.js

```diff
@@ -0,0 +1,61 @@
+const Util = require('util')
+const _ = require('lodash')
+const utils = require('../utils')
+
+function errorMessage (acc, expectedType) {
+  const entityId = _.get(acc, 'reducer.spec.id', 'value')
+  return Util.format(
+    '%s did not pass type check, received: %s of type: %s, expected: %s',
+    entityId,
+    _.truncate(Util.inspect(acc.value, { breakLength: Infinity }), {
+      length: 30
+    }),
+    utils.typeOf(acc.value),
+    expectedType
+  )
+}
```

Comment:
it would be really helpful if the error message says where the type check happened (like whether it was called for `inputType` or `outputType`). however, we can wait and add that when we merge https://github.com/ViacomInc/data-point/issues/149

Reply:
Agreed, and this seems pretty important to include. The difference between if the type is wrong for `inputType` or `outputType` is whether the user has to debug the entire entity or not.

Reply:
is there also a way to log this error message when using a custom reducer for type checking? If those reducers also print `"did not pass type check..."`, it would make it more clear where the error is coming from.